unified_planning/engines/factory.py

```diff
@@ -838,6 +844,31 @@ def Replanner(
             problem=problem,
         )
 
+    def PlanRepairer(
+        self,
+        *,
+        name: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        problem_kind: ProblemKind = ProblemKind(),
+        optimality_guarantee: Optional[Union["OptimalityGuarantee", str]] = None,
+    ) -> "up.engines.engine.Engine":
+        """
+        Returns a plan repairer. There are two ways to call this method:
+        - using 'name' (the name of a plan repairer) and eventually 'params'.
+          e.g. PlanRepairer(name='xxx')
+        - using 'problem_kind' and 'optimality_guarantee'.
+          e.g. PlanRepairer(problem_kind=problem.kind, optimality_guarantee=SOLVED_OPTIMALLY)
+        """
+        if isinstance(optimality_guarantee, str):
+            optimality_guarantee = OptimalityGuarantee[optimality_guarantee]
```

Comment:
I am thinking about what happens when a user writes something that is not a valid `OptimalityGuaranteee`; I guess he gets a `KeyError`. 

To improve the user experience, we might want to encapsulate this code in a try-except block that catches the `KeyError` and re-raises a `UPUsageError` stating that the given string is not a valid `OptimalityGuarantee` and contains the possible `OptimalityGuarantees`.

This is a general comment, not necessarily to do in this PR, as it should be done also in other Operation Modes; but it's such a small change that can also be done here. As you prefer!

Reply:
I think it is better to handle it in a different PR